src/main/java/me/zeroeightsix/kami/module/modules/player/PacketLogger.kt

```diff
@@ -1,100 +1,305 @@
+                            "facing: ${it.packet.facing} " +
+                            "action: ${it.packet.action} ")
+                    }
+                    is CPacketEntityAction -> {
+                        add(Type.CLIENT, it.packet.javaClass.simpleName,
+                            "action: ${it.packet.action} " +
+                            "auxData: ${it.packet.auxData}")
+                    }
+                    is CPacketUseEntity -> {
+                        add(Type.CLIENT, it.packet.javaClass.simpleName,
+                            "action: ${it.packet.action} " +
+                            "hand: ${it.packet.hand} " +
+                            "hitVecX: ${it.packet.hitVec.x} " +
+                            "hitVecY: ${it.packet.hitVec.y} " +
+                            "hitVecZ: ${it.packet.hitVec.z}")
+                    }
+                    is CPacketPlayerTryUseItem -> {
+                        add(Type.CLIENT, it.packet.javaClass.simpleName,
+                            "hand: ${it.packet.hand}")
+                    }
+                    is CPacketConfirmTeleport -> {
+                        add(Type.CLIENT, it.packet.javaClass.simpleName,
+                            "teleportId: ${it.packet.teleportId}")
+                    }
+                    is CPacketChatMessage -> {
+                        if (!ignoreChat) {
+                            add(Type.SERVER, it.packet.javaClass.simpleName,
+                                "message: ${it.packet.message}")
+                        }
+                    }
+                    is CPacketKeepAlive -> {
+                        if (!ignoreKeepAlive) {
+                            add(Type.CLIENT, it.packet.javaClass.simpleName,
+                                "key: ${it.packet.key}")
+                        }
+                    }
+                    else -> {
+                        if (!ignoreUnknown) {
+                            add(Type.CLIENT, it.packet.javaClass.simpleName, "Not Registered in PacketLogger.kt")
+                        }
+                    }
+                }
+                if (logInChat) sendChatMessage(lines.joinToString())
+            }
         }
-    }
 
-    private fun write() {
-        defaultScope.launch(Dispatchers.IO) {
-            try {
-                file.bufferedWriter().use {
-                    lines.forEach(it::write)
-                }
-            } catch (e: Exception) {
-                KamiMod.LOG.error("$chatName Error saving!", e)
+        safeListener<TickEvent.ClientTickEvent> {
+            if (showClientTicks) {
+                if (it.phase == TickEvent.Phase.START) lines.add("Tick Pulse - Realtime: ${sdflog.format(Date())} - Runtime: ${System.currentTimeMillis() - start}ms\n")
+                /* Don't let lines get too big, write periodically to the file */
+                if (player.ticksExisted % 200 == 0 || lines.size >= 1000) write()
```

Comment:
you should be using a coroutine timer for this as well, and the writing should still work even if you don't enable showClientTicks